llvm/lib/CodeGen/SelectionDAG/InstrEmitter.cpp

```diff
@@ -721,9 +721,15 @@ void InstrEmitter::AddDbgValueLocationOps(
     case SDDbgOperand::FRAMEIX:
       MIB.addFrameIndex(Op.getFrameIx());
       break;
-    case SDDbgOperand::VREG:
+    case SDDbgOperand::VREG: {
       MIB.addReg(Op.getVReg());
       break;
+    }
+    case SDDbgOperand::ENTRYVALUEREG: {
+      // We will add our entry value to the expression in our caller.
+      MIB.addReg(Op.getEntryValueReg());
```

Comment:
This LGTM, but we need to start a discussion about upstreaming this.

Reply:
I am fine just upstreaming this. The problem is I need tests to upstream and my tests seem to not be working in OSS as I mentioned due to missing cherry-picks.